OrbitGl/LiveFunctionsDataView.cpp

```diff
@@ -246,3 +280,95 @@ Function& LiveFunctionsDataView::GetFunction(unsigned int a_Row) const {
   CHECK(m_Functions[m_Indices[a_Row]]);
   return *m_Functions[m_Indices[a_Row]];
 }
+
+void LiveFunctionsDataView::GoToFirst(Function& function) {
+  TextBox* box_to_go;
+  TickType minimum_end_time = std::numeric_limits<TickType>::max();
+  std::vector<std::shared_ptr<TimerChain>> chains =
+      GCurrentTimeGraph->GetAllTimerChains();
+  for (auto& chain : chains) {
+    if (!chain) continue;
+    for (TimerChainIterator it = chain->begin(); it != chain->end(); ++it) {
+      TimerBlock& block = *it;
+      for (int i = 0; i < block.size(); i++) {
+        TextBox& box = block[i];
+        if (box.GetTimer().m_FunctionAddress == function.GetVirtualAddress())
+          if (box.GetTimer().m_End < minimum_end_time) {
+            box_to_go = &box;
+            minimum_end_time = box.GetTimer().m_End;
+          }
+      }
+    }
+  }
+  if (minimum_end_time != std::numeric_limits<TickType>::max())
+    GCurrentTimeGraph->Zoom(box_to_go);
+}
+
+void LiveFunctionsDataView::GoToLast(Function& function) {
+  TextBox* box_to_go;
+  TickType maximum_end_time = std::numeric_limits<TickType>::min();
+  std::vector<std::shared_ptr<TimerChain>> chains =
+      GCurrentTimeGraph->GetAllTimerChains();
```

Comment:
With this you are also running through the scheduler and GPU timer chains, which will not contain a timer with a function. Not sure where best to fix this. Maybe introduce another method in TimeGraph that gives you only timer chains that are relevant?

Reply:
I wasn't sure about that. Thanks! I like your suggestion.